pkg/build/controller/factory/factory.go

```diff
@@ -323,13 +323,35 @@ type podLW struct {
 
 // List lists all Pods that have a build label.
 func (lw *podLW) List() (runtime.Object, error) {
-	sel, _ := labels.Parse(buildapi.BuildLabel)
-	return lw.client.Pods(kapi.NamespaceAll).List(sel, fields.Everything())
+	// get builds with new label
+	sel, err := labels.Parse(buildapi.BuildLabel)
+	if err != nil {
+		return nil, err
+	}
+	listNew, err := lw.client.Pods(kapi.NamespaceAll).List(sel, fields.Everything())
+	if err != nil {
+		return nil, err
+	}
+	// FIXME: get builds with old label - remove this when depracated label will be removed
+	selOld, err := labels.Parse(buildapi.DeprecatedBuildLabel)
+	if err != nil {
+		return nil, err
+	}
+	listOld, err := lw.client.Pods(kapi.NamespaceAll).List(selOld, fields.Everything())
+	if err != nil {
+		return nil, err
+	}
+	listNew.Items = append(listNew.Items, listOld.Items...)
```

Comment:
Should not happen, since old pods will have old label and new - the new one. The only time when the two would be there is a human intervention to add the second label.